### PR TITLE
Allow messages returned by the get message body endpoint to just be r…

### PR DESCRIPTION
**Pull request by sonisn1988**
…etrieved in JSON

### CODE REVIEW

File: app/v2/controllers/V2MovementsController.scala [hunk 1]
```diff
@@ -314,6 +319,34 @@ class V2MovementsControllerImpl @Inject() (
...
+    messageSummary: MessageSummary,
+    acceptHeader: String,
+    bodyAndSizeMaybe: Option[BodyAndSize]
+  )(implicit
+    hc: HeaderCarrier
+  ): EitherT[Future, PresentationError, BodyAndContentType] = {
+    def bodyExists(bodyAndSize: BodyAndSize): EitherT[Future, PresentationError, BodyAndContentType] =
+      acceptHeader match {
+        case VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON if bodyAndSize.size > config.smallMessageSizeLimit =>
+          EitherT.leftT[Future, BodyAndContentType](PresentationError.notAcceptableError("Large messages cannot be returned as json"))
```
Comment (drnaylor-oc) [inline app/v2/controllers/V2MovementsController.scala:332]: I think we probably need a better error message than this - we might want to put the size limit or something like that. Might be worth talking to Fergal about it.

File: app/v2/controllers/V2MovementsController.scala [hunk 2]
```diff
@@ -314,6 +319,34 @@ class V2MovementsControllerImpl @Inject() (
...
+        case VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_XML =>
+          EitherT.rightT[Future, PresentationError](
+            BodyAndContentType(MimeTypes.XML, bodyAndSize.body)
+          )
+      }
+    bodyAndSizeMaybe match {
+      case Some(value) => bodyExists(value)
+      case None =>
+        EitherT.leftT[Future, BodyAndContentType](PresentationError.notFoundError(s"Body for message id ${messageSummary.id} does not exist"))
+    }
```
Comment (drnaylor-oc) [inline app/v2/controllers/V2MovementsController.scala:347]: Slight nitpicky but good practice -- can we leave a new line between the function and this? It took me a moment to properly comprehend where the function ended.

I'm a fan of the inner function itself though.

File: app/v2/controllers/V2MovementsController.scala [hunk 3]
```diff
@@ -314,6 +319,34 @@ class V2MovementsControllerImpl @Inject() (
         }
     }
 
+  private def acceptedMessageBody(
```
Comment (drnaylor-oc) [inline app/v2/controllers/V2MovementsController.scala:322]: `formatMessageBody`

**Review by drnaylor-oc — COMMENTED** [review]
Mostly nitpicks, though I haven't completely been through the tests.

File: app/v2/controllers/V2MovementsController.scala [hunk 4]
```diff
@@ -328,7 +328,9 @@ class V2MovementsControllerImpl @Inject() (
     def bodyExists(bodyAndSize: BodyAndSize): EitherT[Future, PresentationError, BodyAndContentType] =
       acceptHeader match {
         case VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON if bodyAndSize.size > config.smallMessageSizeLimit =>
-          EitherT.leftT[Future, BodyAndContentType](PresentationError.notAcceptableError("Large messages cannot be returned as json"))
+          EitherT.leftT[Future, BodyAndContentType](
+            PresentationError.notAcceptableError(s"Message are greater then ${config.smallMessageSizeLimit} bytes and can not retrieve as json")
```
Comment (drnaylor-oc) [inline app/v2/controllers/V2MovementsController.scala:332]: "Messages larger than ${config.smallMessageSizeLimit} bytes cannot be retrieved in Json"

@fmcgrath Can I get some advice on the error message that should go here? We'll return an error if a trader tries to get a message in Json that is larger than the small message size limit (5MB).

Comment (fmcgrath) [reply app/v2/controllers/V2MovementsController.scala:332]: @drnaylor-oc - I agree with your suggestion but I would write Json in ALL CAPS.

Messages larger than ${config.smallMessageSizeLimit} bytes cannot be retrieved in JSON

**Review by fmcgrath — APPROVED** [review]
Text changes look OK.

**Review by drnaylor-oc — APPROVED** [review]
Looks okay now, merge this directly after the regression test counterpart has merged please.

**Pull request merged**